sky/provision/gcp/instance_utils.py

```diff
@@ -75,10 +75,37 @@ def filter(
     def delete_firewall_rule(
         cls,
         project_id: str,
-        firewall_rule_name: str,
+        cluster_name_on_cloud: str,
+        port: str,
     ) -> None:
         raise NotImplementedError
 
+    @classmethod
+    def get_vpc_name(
+        cls,
+        project_id: str,
+        zone: str,
+        instance_name: str,
+    ) -> Optional[str]:
+        raise NotImplementedError
+
+    @classmethod
+    def create_firewall_rule(
+        cls,
+        project_id: str,
+        cluster_name_on_cloud: str,
+        port: str,
+        vpc_name: str,
+    ) -> dict:
+        raise NotImplementedError
+
+
+def _get_firewall_rule_name(
+    cluster_name_on_cloud: str,
+    port: str,
+) -> str:
+    return f'user-ports-{cluster_name_on_cloud}-{port}'
```

Comment:
Is there a length limit for the firewall rule?

Reply:
Yes, it is 63 characters, which is enough since the maximum length of `cluster_name_on_cloud` is only 35 characters.